perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/MailManagerImpl.java

```diff
@@ -1966,6 +2062,23 @@ private ArrayList<String> getFedAuthz() {
 		return fedAuthz;
 	}
 
+	private static String buildUrl(String base, Map<String, String> params, String... pathComponents) {
+		StringBuilder sb = new StringBuilder(base);
+		if (!base.endsWith("/")) {
+			sb.append("/");
+		}
+		Arrays.stream(pathComponents)
+			.map(String::trim) // remove spaces
+			.map(s -> s.endsWith("/") ? s + "/" : s) // all path components need to end with "/"
```

Comment:
Shouldn't the cases be switched ? I believe this way we add extra "/" after existing one instead of adding it when it is missing.